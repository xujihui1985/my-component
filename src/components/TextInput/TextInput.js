import React from 'react';
import PropTypes from 'prop-types';
import Label from '../Label';
import cns from 'classnames';

function TextInput({
  htmlID,
  name,
  label,
  type = "text",
  required = false,
  onChange,
  placeholder,
  value,
  error,
  ...rest
}) {
  return (
    <div style={{ marginBottom: 16 }}>
      <Label
        htmlFor={htmlID}
        label={label}
        required={required}
      />
      <input
        id={htmlID}
        type={type}
        name={name}
        placeholder={placeholder}
        value={value}
        onChange={onChange}
        className={cns({
          'textinput__input--state-error': error
        })}
        {...rest}
      />
      {
        error &&
        (
          <div
            className={cns({ 'textinput__error': true })}
          >
            {error}
          </div>
        )
      }
    </div>
  );
}


export default TextInput;