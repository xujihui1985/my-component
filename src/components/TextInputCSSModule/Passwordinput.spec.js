import React from 'react';
import renderer from 'react-test-renderer';
import TextInput from './';
import { shallow } from 'enzyme';

test('show error', () => {
  const tree = renderer.create(
    <TextInput
      htmlID="registration-form-email"
      name="email"
      onChange={() => { }}
      label="Email"
      value={"email"}
      error={"error"}
      required
    />
  );
  expect(tree).toMatchSnapshot();
});

test('toggles input type when show/hide password checked', () => {
  const wrapper = shallow(
    <TextInput
      htmlID="test"
      name="test"
      label="test"
      value=""
      onChange={() => { }}
      showVisibilityToggle
    />
  );
  expect(wrapper.find({ type: 'text' })).toHaveLength(1);

});
