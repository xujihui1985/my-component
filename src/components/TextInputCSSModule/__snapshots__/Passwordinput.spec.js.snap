// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`show error 1`] = `
<div
  className={undefined}
>
  <label
    htmlFor="registration-form-email"
    style={
      Object {
        "display": "block",
      }
    }
  >
    Email
     
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
       *
    </span>
  </label>
  <input
    className="inputError"
    id="registration-form-email"
    name="email"
    onChange={[Function]}
    placeholder={undefined}
    type="text"
    value="email"
  />
  <div
    className={undefined}
  >
    error
  </div>
</div>
`;
